packages/aws-cdk-lib/aws-route53-targets/lib/appsync-target.ts

```diff
@@ -0,0 +1,23 @@
+import { CloudFrontTarget } from './cloudfront-target';
+import { GraphqlApi } from '../../aws-appsync';
+import {
+  AliasRecordTargetConfig,
+  IAliasRecordTarget,
+  IHostedZone,
+  IRecordSet,
+} from '../../aws-route53';
+
+/**
+ * Defines an AppSync Graphql API as the alias target. Requires that the domain
+ * name will be defined through `GraphqlApiProps.domainName`.
+ */
+export class AppSync implements IAliasRecordTarget {
```

Comment:
nit: I would prefer to name it as `AppSyncTarget`.

Reply:
I went with `AppSync` rather than `AppSyncTarget` to match the `ApiGateway` route53 target, but I now realise that all the other targets are suffixed with `Target` so have done that for this too: 400f6bc5f3